ProcessLib/BoundaryCondition/NonuniformDirichletBoundaryCondition.h

```diff
@@ -0,0 +1,120 @@
+/**
+ * \file
+ *
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include "BoundaryCondition.h"
+
+#include "MeshLib/PropertyVector.h"
+#include "NumLib/DOF/DOFTableUtil.h"
+#include "NumLib/DOF/LocalToGlobalIndexMap.h"
+#include "NumLib/IndexValueVector.h"
+#include "ProcessLib/Parameter/Parameter.h"
+#include "ProcessLib/Utils/ProcessUtils.h"
+
+namespace ProcessLib
+{
+class NonuniformDirichletBoundaryCondition final : public BoundaryCondition
+{
+public:
+    NonuniformDirichletBoundaryCondition(
+        // int const bulk_mesh_dimension,
+        std::unique_ptr<MeshLib::Mesh>
+            boundary_mesh,
+        MeshLib::PropertyVector<double> const& values,
+        std::size_t const bulk_mesh_id,
+        MeshLib::PropertyVector<std::size_t> const& mapping_to_bulk_nodes,
+        NumLib::LocalToGlobalIndexMap const& dof_table_bulk,
+        int const variable_id_bulk,
+        int const component_id_bulk)
+        : _bulk_mesh_id(bulk_mesh_id),
+          _values(values),
+          _boundary_mesh(std::move(boundary_mesh)),
+          _mapping_to_bulk_nodes(mapping_to_bulk_nodes),
+          _dof_table_bulk(dof_table_bulk),
+          _variable_id_bulk(variable_id_bulk),
+          _component_id_bulk(component_id_bulk)
+    {
+        if (_variable_id_bulk >=
+                static_cast<int>(_dof_table_bulk.getNumberOfVariables()) ||
+            _component_id_bulk >= _dof_table_bulk.getNumberOfVariableComponents(
+                                      _variable_id_bulk))
+        {
+            OGS_FATAL(
+                "Variable id or component id too high. Actual values: (%d, "
+                "%d), "
+                "maximum values: (%d, %d).",
+                _variable_id_bulk, _component_id_bulk,
+                _dof_table_bulk.getNumberOfVariables(),
+                _dof_table_bulk.getNumberOfVariableComponents(
+                    _variable_id_bulk));
+        }
+    }
+
+    void preTimestep(const double /*t*/) override {}
+
+    void getEssentialBCValues(
+        const double /*t*/,
```

Comment:
Too bad, that time dependent non-uniform BCs don't work, yet. TODO for the future.

Reply:
this would be very important. :+1: from my side for this.

Reply:
Noted in https://github.com/ufz/ogs/issues/1915